src/ruleset/v2/functions/messageExamples-spectral-rule-v2.ts

```diff
@@ -0,0 +1,138 @@
+import { RuleDefinition, createRulesetFunction } from '@stoplight/spectral-core';
+
+import type { JsonPath } from '@stoplight/types';
+import type { IFunctionResult } from '@stoplight/spectral-core';
+import type { v2 } from 'spec-types';
+import type { Parser } from 'parser';
+import type { DetailedAsyncAPI } from 'types';
+import { ParseSchemaInput, getDefaultSchemaFormat, getSchemaFormat, parseSchema } from '../../../schema-parser';
+import { createDetailedAsyncAPI } from '../../../utils';
+import { getMessageExamples, validate } from './messageExamples';
+
+export function asyncApi2MessageExamplesParserRule(parser: Parser): RuleDefinition {
+  return {
+    description: 'Examples of message object should validate against the "payload" and "headers" schemas.',
```

Comment:
We agree this rule now only are `Examples of message object when using explicit schema format should validate correctly against payloads` right? 🤔